runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -0,0 +1,65 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Message;
+
+import org.corfudb.runtime.CorfuOptions;
+import org.corfudb.runtime.collections.CorfuTable.Index;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Optional;
+
+/**
+ * Created by hisundar on 2019-08-12.
+ */
+public class ProtobufIndexer implements CorfuTable.IndexRegistry<Message, CorfuRecord<Message, Message>> {
+
+    private HashMap<String,
+            CorfuTable.Index<Message, CorfuRecord<Message, Message>, ? extends Comparable<?>>>
+            indices = new HashMap<>();
+
+    ProtobufIndexer(Message payloadSchema) {
+        payloadSchema.getDescriptorForType().getFields().forEach(this::registerIndices);
+    }
+
+    private <T extends Comparable<T>> CorfuTable.Index<Message, CorfuRecord<Message, Message>, ? extends Comparable<?>>
```

Comment:
Can you please create a particular (domain) interface (that will extend Comparable<T>) and the interface name can explain what is that. Because right now it's completely unclear why Comparable used here

Reply:
But won't that mean that I have to change CorfuTable's Indexer too - that's complicated as it is.